politeiad/backendv2/tstorebe/plugins/comments/comments.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2020-2021 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package comments
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+
+	"github.com/decred/politeia/politeiad/api/v1/identity"
+	backend "github.com/decred/politeia/politeiad/backendv2"
+	"github.com/decred/politeia/politeiad/backendv2/tstorebe/plugins"
+	"github.com/decred/politeia/politeiad/plugins/comments"
+)
+
+var (
+	_ plugins.PluginClient = (*commentsPlugin)(nil)
+)
+
+// commentsPlugin is the tstore backend implementation of the comments plugin.
+//
+// commentsPlugin satisfies the plugins.PluginClient interface.
+type commentsPlugin struct {
+	sync.RWMutex
+	tstore plugins.TstoreClient
+
+	// dataDir is the comments plugin data directory. The only data
+	// that is stored here is cached data that can be re-created at any
+	// time by walking the trillian trees.
+	dataDir string
+
+	// identity contains the full identity that the plugin uses to
+	// create receipts, i.e. signatures of user provided data that
+	// prove the backend received and processed a plugin command.
+	identity *identity.FullIdentity
+
+	// Plugin settings
+	commentLengthMax uint32
+	voteChangesMax   uint32
+}
+
+// Setup performs any plugin setup that is required.
+//
+// This function satisfies the plugins.PluginClient interface.
+func (p *commentsPlugin) Setup() error {
+	log.Tracef("comments Setup")
+
+	return nil
+}
+
+// Cmd executes a plugin command.
+//
+// This function satisfies the plugins.PluginClient interface.
+func (p *commentsPlugin) Cmd(treeID int64, token []byte, cmd, payload string) (string, error) {
+	log.Tracef("comments Cmd: %v %x %v", treeID, token, cmd)
+
+	switch cmd {
+	case comments.CmdNew:
+		return p.cmdNew(treeID, token, payload)
+	case comments.CmdEdit:
+		return p.cmdEdit(treeID, token, payload)
+	case comments.CmdDel:
+		return p.cmdDel(treeID, token, payload)
+	case comments.CmdVote:
+		return p.cmdVote(treeID, token, payload)
+	case comments.CmdGet:
+		return p.cmdGet(treeID, token, payload)
+	case comments.CmdGetAll:
+		return p.cmdGetAll(treeID, token)
+	case comments.CmdGetVersion:
+		return p.cmdGetVersion(treeID, token, payload)
+	case comments.CmdCount:
+		return p.cmdCount(treeID, token)
+	case comments.CmdVotes:
+		return p.cmdVotes(treeID, token, payload)
+	case comments.CmdTimestamps:
+		return p.cmdTimestamps(treeID, token, payload)
+	}
+
+	return "", backend.ErrPluginCmdInvalid
+}
+
+// Hook executes a plugin hook.
+//
+// This function satisfies the plugins.PluginClient interface.
+func (p *commentsPlugin) Hook(treeID int64, token []byte, h plugins.HookT, payload string) error {
+	log.Tracef("comments Hook: %v %x %v", plugins.Hooks[h], token, treeID)
+
+	return nil
+}
+
+// Fsck performs a plugin filesystem check.
+//
+// This function satisfies the plugins.PluginClient interface.
+func (p *commentsPlugin) Fsck(treeIDs []int64) error {
```

Comment:
We need to rethink this. If the cache can become corrupt it probably needs to be redone. Let's debate options.

Reply:
We've punted on this until we replace the cache with reddis/memcached.